backend/server/src/db/model/drv.rs

```diff
@@ -190,13 +205,80 @@ mod tests {
     }
 }
 
-#[derive(Clone, Debug, FromRow)]
+#[derive(Debug, Clone, FromRow)]
 pub struct Drv {
     /// Derivation identifier.
     pub derivation: DrvId,
 
     /// to reattempt the build (depending on the interruption kind).
     pub system: String,
+
+    pub required_system_features: Option<String>,
+}
+
+#[derive(Debug, Deserialize)]
+struct DrvOutput {
+    // nix derivaiton show always structures the output as:
+    // { ${drv}: { ... } }
+    #[serde(flatten)]
+    drvs: HashMap<String, DrvAttrs>,
+}
+
+#[derive(Debug, Deserialize)]
+struct DrvAttrs {
+    pub env: DrvInfo,
+}
+
+#[derive(Debug, Deserialize)]
+struct DrvInfo {
+    /// to reattempt the build (depending on the interruption kind).
+    pub system: String,
+
+    #[serde(rename = "requiredSystemFeatures")]
+    pub required_system_features: Option<String>,
+}
+
+impl Drv {
+    pub fn full_drv_path(&self) -> String {
+        format!("/nix/store/{}", &self.derivation.0)
+    }
+
+    /// Calls `nix derivation show` to retrieve system and requiredSystemFeatures
+    pub async fn fetch_info(drv_path: &str) -> anyhow::Result<Self> {
+        let drv_output = drv_output(drv_path).await?;
+        Ok(Drv {
+            derivation: DrvId::new(drv_path.to_string()),
+            system: drv_output.system,
+            required_system_features: drv_output.required_system_features,
+        })
+    }
+}
+
+pub fn strip_store_prefix(drv_path: String) -> String {
+    drv_path
+        .strip_prefix("/nix/store/")
+        .unwrap_or(&drv_path)
+        .to_string()
+}
+
+/// Do `nix derivation show` but filter for the things we care about
+async fn drv_output(drv_path: &str) -> anyhow::Result<DrvInfo> {
+    debug!("Fetching derivation information, {:?}", &drv_path);
+    let output = Command::new("nix")
+        .args(["derivation", "show", drv_path])
+        .output()
+        .await?
+        .stdout;
+    let str = String::from_utf8(output)?;
+    let drv_output: DrvOutput = serde_json::from_str(&str)?;
+    // FIXME: get rid of expect
+    let drv_info = drv_output
+        .drvs
+        .into_iter()
+        .next()
+        .expect("Invalid derivation show information")
```

Comment:
```suggestion
        .context("Invalid derivation show information")?
```
Should work with an `Option` too I think